src/main/java/formflow/library/FileController.java

```diff
@@ -144,26 +172,26 @@ RedirectView delete(
       Optional<Submission> maybeSubmission = submissionRepositoryService.findById(submissionId);
 
       if (maybeSubmission.isEmpty()) {
-        log.error(String.format("Session %d does not exist", submissionId));
+        log.error(String.format("Submission %s does not exist", submissionId.toString()));
         return new RedirectView("/error");
       }
 
-      Optional<UserFile> maybeFile = uploadedFileRepositoryService.findById(fileId);
+      Optional<UserFile> maybeFile = userFileRepositoryService.findById(fileId);
       if (maybeFile.isEmpty()) {
         log.error(String.format("File with id %s may have already been deleted", fileId));
         return new RedirectView("/error");
       }
 
       UserFile file = maybeFile.get();
-      if (!submissionId.equals(file.getSubmission_id().getId())) {
-        log.error(String.format("Submission %d does not match file %s's submission id %d", submissionId, fileId,
-            file.getSubmission_id().getId()));
+      if (!submissionId.equals(file.getSubmissionId().getId())) {
```

Comment:
As Chibuisi mentioned, this is really confusing.  The naming is definitely misleading:`getSubmissionId` returning the `Submission`. Can we make this easier to work work with or are we stuck with this because of DB magic?  

Can we call variable `submissionId` `submission` instead, in the UserFile.java code?  That might work, since we specify the column name in an attribute - so it knows which column to grab. Perhaps the variable name in that file doesn't carry any semantic significance?